### DISCUSSION TITLE
蓝牙信号接收原理

### DISCUSSION THREAD

**Discussion by TTsXDU** [wh201906/SerialTest#59] [Q&A]
我想知道这款上位机是如何接收外部蓝牙客户端的蓝牙信号的，是如何搜索到的，在程序里是怎么实现的？蓝牙连接程序的编写思路是什么呢？我想了解的透彻一些，但代码看不太懂，博主能详细说下实现的原理吗？感激不尽

**Comment by wh201906** [comment]
这个上位机主要是靠调用Qt的库函数来完成蓝牙连接。Qt的蓝牙库将不同平台下的蓝牙API整合在一起，形成统一的函数接口，方便开发者去编写一套可在多个平台上运行的统一代码。具体怎么连接外部设备取决于各个平台的具体实现，可能有所不同，但只要都符合蓝牙规范即可。这方面你可以去查阅蓝牙的数据手册。
至于上位机的程序流程你可以从“连接”页面当中点击“打开”按钮对应的函数一路跟下去，结合[doc.qt.io](https://doc.qt.io/qt-5/qtbluetooth-index.html)上有关蓝牙的文档去分析，就能理解了。

**Reply by TTsXDU** [reply]
哇好的，感谢博主指点
